src/node_url.cc

```diff
@@ -92,6 +92,15 @@ class URLHost {
   Value value_;
   HostType type_ = HostType::H_FAILED;
 
+  inline void Reset() {
+    using string = std::string;
+    switch (type_) {
+      case HostType::H_DOMAIN: value_.domain.~string(); break;
+      case HostType::H_OPAQUE: value_.opaque.~string(); break;
+      default: break;
+    }
```

Comment:
After resettig the `value_`, can you reset the `type_` as well by setting it to `HostType::H_FAILED`?

Reply:
Just add this line `type_ = HostType::H_FAILED;` after switch statement?